### PR TITLE
Defect/resume questions

### PR DESCRIPTION
**Pull request by JoeKer1**

### CODE REVIEW

File: src/app/route-guards/resume-old-assessment.guard.ts [hunk 1]
```diff
@@ -23,21 +23,48 @@ export class ResumeOldAssessmentGuard implements CanActivate {
...
   canActivate(
     next: ActivatedRouteSnapshot,
     state: RouterStateSnapshot
   ): Observable<boolean> {
     if (this.dataService.GetAppJustLoaded()) {
       this.dataService.SetAppLoaded();
       return this.dataService.GetConfig().flatMap(config => {
-        this.router.navigate([""]);
-        return this.returnObs(true);
+        const assessmentId = this.cookieService.get("AssessmentID");
```
Comment (Doug-Shannon) [inline src/app/route-guards/resume-old-assessment.guard.ts]: are you purposefully reinstating the resume functionality?

File: src/app/services/assessment.service.ts [hunk 2]
```diff
@@ -106,22 +109,18 @@ export class AssessmentService {
...
   }
 
-  private QuestionsArrayAssesment(questions: Question[]): Assessment {
+  private QuestionsArrayAssessment(questions: Question[]): Assessment {
     const assessment = new Assessment();
     assessment.responses = new Array<WfResponse>();
 
-    //dont dot chain the following 2 functions, it will not work.
-    questions.sort(this.sortQuestionsRandomly);
+    questions = _.shuffle(questions);
```
Comment (Doug-Shannon) [inline src/app/services/assessment.service.ts]: good call

Comment (Doug-Shannon) [reply src/app/services/assessment.service.ts]: but, that function was more of a placeholder for when we know how they want to sort (I think its fine to remove for now)

File: src/app/services/data.service.ts [hunk 3]
```diff
@@ -42,57 +43,76 @@ export class DataService {
...
       assessment.documentid !== undefined
         ? assessment.documentid
         : this.documentDB.createId();
-
+    assessment.documentid = assessmentDocumentid;
     const jsonObject = this.prepareDocForAFS(assessment);
-
-    const documentRef = this.documentDB
+    const batch = this.documentDB.firestore.batch();
+    const documentRef = this.documentDB.firestore
```
Comment (Doug-Shannon) [inline src/app/services/data.service.ts]: this changes the object type to DocumentReference and is no longer an AngularFirestoreDocumentReference - I think its fine, just an FYI

File: src/app/route-guards/resume-old-assessment.guard.ts (hunk 1 above)
Comment (JoeKer1) [reply src/app/route-guards/resume-old-assessment.guard.ts]: yarp

File: src/app/services/assessment.service.ts (hunk 2 above)
Comment (JoeKer1) [reply src/app/services/assessment.service.ts]: ahh yeah. I suppose we can rip out lodash and do our own function when we know more.

File: src/app/services/data.service.ts (hunk 3 above)
Comment (JoeKer1) [reply src/app/services/data.service.ts]: yeah i wonder if they will let batching use AngularFirestoreDocumentReferences in the future. It would be nice.

**Pull request merged**